packages/trader/src/Modules/CFD/Containers/cfd-password-modal.tsx

```diff
@@ -453,24 +588,24 @@ const CFDServerModalWarning = ({ show_warning = true, platform }) => {
     );
 };
 
-const CFDServerForm = ({ ...props }) => {
+const CFDServerForm = ({ ...props }: TCFDServerFormProps) => {
     const available_servers = React.useMemo(() => {
         return [...props.mt5_trading_servers]
             .map(server => {
                 // Transform properties to support radiogroup
                 return {
                     ...server,
                     ...{
-                        label: `${server.geolocation.region} ${
-                            server.geolocation.sequence === 1 ? '' : server.geolocation.sequence
+                        label: `${server.geolocation?.region} ${
+                            server.geolocation?.sequence === 1 ? '' : server.geolocation?.sequence
                         } ${server.disabled ? `(${server.message_to_client})` : ''}`,
                         value: server.id,
                         disabled: server.disabled,
                     },
                 };
             })
-            .sort((a, b) => (a.recommended ? a : b))
-            .sort((a, b) => a.disabled - b.disabled);
+            .sort((a, _b) => (a.recommended ? -1 : 1))
+            .sort((a, b) => (a.disabled ? a.disabled : 0) - (b.disabled ? b.disabled : 0));
```

Comment:
I added the check to this callback in order to avoid a TS error saying that the 'disabled' property can be not only '0 | 1' but also 'undefined' according to its type in api-types. The way I implemented it is based on an assumption that the design is to make an item, which has 0 | 'undefined' value of the 'disabled' property, go before the next item.